pyomo/solvers/tests/checks/test_mosek.py

```diff
@@ -0,0 +1,139 @@
+import pyutilib.th as unittest
+from pyomo.opt import *
+from pyomo.environ import *
+import sys
+
+try:
+    import mosek
+    mosek_available = True
+except ImportError:
+    mosek_available = False
+
+diff_tol = 1e-3
+
+
+class MosekDirectTests(unittest.TestCase):
+
+    def setUp(self):
+        self.stderr = sys.stderr
+        sys.stderr = None
+
+    def tearDown(self):
+        sys.stderr = self.stderr
+
+    @unittest.skipIf(not mosek_available,
+                     "The 'mosek' python bindings are not available")
```

Comment:
Generally, I think we should rely on `SolverFactory("mosek", solver_io='python').available()` instead of the import test.